storage/storage.js

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const path = require('path');
+const {
+	BaseEntity,
+	Account,
+	Delegate,
+	Block,
+	Transaction,
+} = require('./entities');
+const PgpAdapter = require('./adapters/pgp_adapter');
+
+class Storage {
+	constructor(options, logger) {
+		this.options = options;
+		this.logger = logger;
+
+		if (typeof Storage.instance === 'object') {
+			return Storage.instance;
+		}
+
+		this.isReady = false;
+
+		Storage.instance = this;
+
+		Storage.instance.BaseEntity = BaseEntity;
+		Storage.instance.BaseEntity.adapter = null;
+	}
+
+	/**
+	 * @return Promise
+	 */
+	bootstrap() {
+		const adapter = new PgpAdapter(
+			Object.assign({}, this.options, {
+				inTest: process.env.NODE_ENV === 'test',
+				sqlDirectory: path.join(path.dirname(__filename), './sql'),
+				logger: this.logger,
+			})
+		);
+
+		return adapter.connect().then(status => {
+			if (status) {
```

Comment:
prefer early return

Reply:
What do you mean? It's always returning status here.

Reply:
ah nvm, my bad ='(